# Disables OpenAL Soft's null backend since it fails to compile for PlayStation 3. Don't worry, we don't need the null backend.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -881,7 +881,7 @@ set(ALC_OBJS  ${ALC_OBJS}
     # Default backends, always available
     alc/backends/loopback.cpp
     alc/backends/loopback.h
-    alc/backends/null.cpp
+
     alc/backends/null.h
 )
 
diff --git a/alc/alc.cpp b/alc/alc.cpp
index af8ff55d..0319a486 100644
--- a/alc/alc.cpp
+++ b/alc/alc.cpp
@@ -251,7 +251,7 @@ BackendInfo BackendList[] = {
     { "sdl2", SDL2BackendFactory::getFactory },
 #endif
 
-    { "null", NullBackendFactory::getFactory },
+    { "null", nullptr },
 #ifdef HAVE_WAVE
     { "wave", WaveBackendFactory::getFactory },
 #endif
@@ -1289,7 +1289,7 @@ void alc_initconfig(void)
     {
         if(PlaybackFactory && CaptureFactory)
             return;
-
+        if (!std::strcmp(backend.name, "null")) return;
         BackendFactory &factory = backend.getFactory();
         if(!factory.init())
         {
